src/json/program/ProgramTriggerVariableContext.schema.ts

```diff
@@ -0,0 +1,396 @@
+  type: "object",
+  title: "Program Variable Context Schema",
+  description:
+    "Defines the context used for evaluating externally defined inputs (e.g. JSONata)",
+  properties: {
+    type: {
+      type: "string",
+      title: "Trigger Type",
+      description:
+        "The reason the program was triggered (e.g. referral created, user changed something, etc.)",
+      enum: [
+        "AFTER_USER_CREATED_OR_UPDATED",
+        "SCHEDULED",
+        "REWARD_SCHEDULED",
+        "REFERRAL",
+        "AFTER_USER_EVENT_PROCESSED"
+      ]
+    }
+  }
+};
+
+const programTriggerVariableContextSchema2: JSONSchema6 = {
+  $schema: "http://json-schema.org/draft-06/schema#",
+  type: "object",
+  title: "Program Variable Context Schema",
+  description:
+    "Defines the context used for evaluating externally defined inputs (e.g. JSONata)",
+  properties: {
+    type: {
+      type: "string",
+      title: "Trigger Type",
+      description:
+        "The reason the program was triggered (e.g. referral created, user changed something, etc.)",
+      enum: [
+        "AFTER_USER_CREATED_OR_UPDATED",
+        "SCHEDULED",
+        "REWARD_SCHEDULED",
+        "REFERRAL",
+        "AFTER_USER_EVENT_PROCESSED"
+      ]
+    }
+  },
+  dependencies: {
+    type: {
+      oneOf: [
+        {
+          $ref: "#/definitions/afterUserCreatedOrUpdatedTriggerContext"
+        },
+        {
+          $ref: "#/definitions/rewardScheduledProgramTriggerContext"
+        },
+        {
+          $ref: "#/definitions/scheduledProgramTriggerContext"
+        },
+        {
+          $ref: "#/definitions/referralTriggerContext"
+        },
+        {
+          $ref: "#/definitions/afterUserEventProcessedTriggerContext"
+        }
+      ]
+    }
+  },
+  definitions: {
+    afterUserCreatedOrUpdatedTriggerContext: {
+      $ref: "AfterUserCreatedOrUpdatedTriggerContext.schema.json#"
+    },
+    rewardScheduledProgramTriggerContext: {
+      $ref: "RewardScheduledProgramTriggerContext.schema.json#"
+    },
+    scheduledProgramTriggerContext: {
+      $ref: "ScheduledProgramTriggerContext.schema.json#"
+    },
+    referralTriggerContext: {
+      $ref: "ReferralTriggerContext.schema.json#"
+    },
+    afterUserEventProcessedTriggerContext: {
+      $ref: "AfterUserEventProcessedTriggerContext.schema.json#"
+    }
+  }
+};
+
+const baseUserContextFields: JSONSchema6["properties"] = {
+  id: {
+    type: "string",
+    title: "User Id"
+  },
+  accountId: {
+    type: "string",
+    title: "Account Id"
+  },
+  firstName: {
+    type: ["string", "null"],
+    title: "First Name"
+  },
+  lastName: {
+    type: ["string", "null"],
+    title: "Last Name"
+  },
+  imageUrl: {
+    type: ["string", "null"],
+    title: "Image Url"
+  },
+  email: {
+    type: ["string", "null"],
+    title: "Email Address"
+  },
+  cookieId: {
+    type: ["string", "null"],
+    title: "Cookie Id"
+  },
+  locale: {
+    type: ["string", "null"],
+    title: "User Locale"
```

Comment:
The `user` part is implied I think

```suggestion
    title: "Locale"
```